python/TestHarness/testers/RunApp.py

```diff
@@ -280,6 +276,11 @@ def processResults(self, moose_dir, options, output):
                 refactor testFileOutput and processResults.
         """
         output += self.testFileOutput(moose_dir, options, output)
-        output += self.testExitCodes(moose_dir, options, output)
+
+        # TODO: Do the TODO! and refactor testFileOutput, testExitCodes, processResults.
+        reason = self.testExitCodes(moose_dir, options, output)
+        if reason != '':
+            self.setStatus(self.fail, reason)
+            output += "\n\nExit Code: " + str(self.exit_code)
```

Comment:
I thought we weren't going to try to refactor this just yet. This isn't going to work like you want because processResults() is a virtual method. It's overridden in most of the Tester objects. Many of the Tester objects are assuming that TestExitCodes() is going to return extra stuff to append to the output (the existing behavior) but now you are returning reason. 

If you are going to refactor, you have to change every Tester object, or you can just do that in a separate PR.